 ### Graded Assignment on Descrptive Statistics

#### Section A

In [1]:
import pandas as pd
import numpy as np
from scipy.stats import gmean

_________

#### Question 1

The average selling price of a food item in a restaurant is Rs 250.00 and the standard deviation of the price is Rs 47.00. 
The median value is an indication of the central value of the data and isclose to the mean of Rs 202.00. If you know these 
details, will you become a regular customer of the restaurant?

### Answer 1

1. Median is middle price price of food item i.e Rs 202
2. Mean of food item is 250 but as we know that mean is not provided the accurate or may not be close the price of each food item it may be some food item price is 20 or 300 so we can not decide the price of food item by the mean.
3. After compare the mean with Standard Deviation(SD). I conculde that SD have less variability it means the price of food item is close to the mean. 
3. Now, I have a price idea of food item. But, Customer will become a regular or not its depend on various factor not only price of food item like
a) Taste of food
b) Income of the customer
c) Expenditure on food on the daily basis.
d) Customer get more satisfaction when he spent money on food item in a resturant as compared to other resturant or not.
we can not predict or analysis the customer behaviour by the price of food only. It is depend on above mentioned factor for
analysing the customer will becomer regular or not.
so data is insufficient in my point of view.

-----------

#### Question 2

In a residential locality, the average size of a house is 2224 square feet with a standard deviation of 248 square feet 
which indicates that the distribution of size is fairly the sameacross the data and the median value of the house is 2220 
square feet. Does the locality have bigger or smaller-sized houses?

### Answer 2

In [ ]:
1. As we can see that the median is closer to the mean and if,
2. we compare the mean with the SD than, SD deviation is small.
3. which shows that the data of the house's size have less variability.
4. so, we can say that the size of mostly house's size is closer to the mean.
But we can not identify that the locality have bigger or not.  to identify this we need one more locality data for the
comparison then we can identify that locality is bigger or not.

-----------

#### Question 3

You want to invest in real estate, and you have the data belonging to locality X and Y. Some of the house prices in locality
X are 2.00 L,10.00 L, 8.00 L, and 12.00 L and the prices of houses in Y are 2.5 L, 9.50 L, 8.15 L, and 12.50 L. Conduct a 
statistical measure to decide which locality is best for investment.

### Answer 3

In [7]:
x = [2.00, 10.00, 8.00, 12.00]
y = [2.5, 9.50, 8.15, 12.50]
data = np.array(x)
data_1 = np.array(y)

In [8]:
# calculate mean of locality X and Y
d_mean = data.mean()
d1_mean = data_1.mean()
# calculate standard deivation of locality X and Y
d_std = data.std()
d1_std = data_1.std()

In [49]:
# print result
print("x locality result:")
print("mean:",d_mean)
print("standard deviation:",round(d_std,2))
#print result of y locality
print("y locality result:")
print("mean:",d1_mean)
print("standard deviation:",round(d1_std,2))

x locality result:
mean: 8.0
standard deviation: 3.74
y locality result:
mean: 8.1625
standard deviation: 3.63


1. Mean of x locality is 8 & the mean of y locality is 8.16 which is describe that the price of locality y is little high as compared to locality y.
2. SD of x locality is 3.7 approx and SD of locality Y is 3.6 which shows that the SD, X locality is higher than the Y locality. it means that the x locality has more variability means widely spread as compared to Y locality and vise-versa. 
3. so, in my point of view locality Y is best for the investment.( But other factors remain constant becuse the investment is depend on other factors like location etc.)

----------

#### Question 4

During a survey, the ages of 80 patients, infected by COVID and admitted to one of the city
hospitals, were recorded. The collected data is represented in the less-than-cumulative
frequency distribution table. Read the questions below the table and provide your answers

### Answer 4

In [32]:
survey = pd.DataFrame({'Age': ['5-15','15-25','25-35','35-45','45-55','55-65'],
                       'Patient': [6,11,21,23,14,5]})
survey

Age  Patient
0   5-15        6
1  15-25       11
2  25-35       21
3  35-45       23
4  45-55       14
5  55-65        5

In [14]:
# a. Which class interval has the highest frequency?
max_patient = survey['Patient'].max()
class_interval_max_frequency = survey.loc[survey["Patient"]==max_patient]
print(class_interval_max_frequency)

     Age  Patient
3  35-45       23


In [15]:
# b. Which age was affected the least?
min_frequency_patient = survey["Patient"].min()
least_age_affected = survey.loc[survey["Patient"] == min_frequency_patient, "Age"]
print(least_age_affected)

5    55-65
Name: Age, dtype: object


In [16]:
# c. How many patients of age 45 years and above were admitted?
patient_admitted = survey.loc[survey["Age"]>="45"]
a = survey.iloc[0,0] # because age group in string and loc will considered this condition character by character
Above_45_patient_admitted = patient_admitted.loc[patient_admitted["Age"] != a,"Patient"].sum()
print(Above_45_patient_admitted)

19


In [24]:
# d. Which is the modal class interval in the above dataset?
max_frequency = survey["Patient"].max()
modal_class = survey.loc[survey["Patient"] == max_frequency ,"Age"]
print("modal class of survey report:", modal_class)

modal class of survey report: 3    35-45
Name: Age, dtype: object


In [42]:
# e. What is the median class interval of age?
survey["cum"] = survey["Patient"].cumsum()
nth_value = (survey["Patient"].sum())/len(survey)
median_CI_age = survey.loc[survey["cum"]>=a2,"Age"].min()
median_CI_age


'15-25'

-----------

#### Question 5
Assume, you are a trader and have invested over the years for your retirement. Lately, with the fluctuating economical condition, you are worried about the average return on investment. What average method is appropriate to compute the averagereturn for the data given in the table?

### Answer 5

In [43]:
# geometric mean is suitable to calculate the average return. 

trader = pd.DataFrame({"Year":[2015,2016,2017,2018,2019,2020],
                        "Return":[0.36,0.23,-0.48,-0.30,0.15,0.31],
                        "Asset Price": [5000,6400,7890,9023,4567,3890],})
trader['gmean'] = trader['Return']+1
gmean = trader['gmean'].prod()
average_return = gmean**(1/len(trader)) -1
print("{:.2%}".format(average_return))

-1.43%


# or

In [44]:
returns = [0.36, 0.23, -0.48, -0.30, 0.15, 0.31]
gem = np.prod(1+np.array(returns))**(1/len(returns)) -1
print("{:.2%}".format(gem))

-1.43%


--------

#### Question 6

Suppose you have been told to measure the height of all men on the earth. What strategy would you use?

### Answer

As we know that it is difficult to calculate the height of all men on the earth so we will calculate the height of all men on 
the earth by sampling variance method. In this method we collect the some data of men's height through the survey then, we will
use sampling method. for. e.g.
i have collected some data of men's height through the survey men_height = 155,165,169,170,172,165,150,155 then we can used the
sample variance which help me to identify the men's height on the earth.

In [6]:
men_height = [155,165,169,170,172,165,150,155]
men_height_1 = np.array(men_height)
men_height_1

array([155, 165, 169, 170, 172, 165, 150, 155])

In [7]:
mean_men_height = men_height_1.mean()
s_variance_men = (men_height_1 - mean_men_height)**2
s_variance_men_height = (s_variance_men).sum()/(len(men_height_1) -1)
s_std_men_height = (s_variance_men_height)**0.5
print('variance:',s_variance_men_height)
print('standard deviation:',s_std_men_height)

variance: 67.125
standard deviation: 8.19298480408697


-------

#### Question 7
##### Calculate the z score of the following numbers

### Answer 7

In [9]:
x = pd.DataFrame({"Number":[4.5,6.2,7.3,9.1,10.4,11]})
mean = "{:.3}".format(x["Number"].mean())
mean = float(mean) # i am converting mean in int from string after formating
std = x["Number"].std()
std = float(std)
z_score_1 = round((x['Number']-mean)/std,2)
z_score_1

0   -1.42
1   -0.74
2   -0.31
3    0.40
4    0.92
5    1.16
Name: Number, dtype: float64

---------
--------

# section 2

In [78]:
df = pd.read_excel(r"C:\Users\hp\Desktop\SAURABH\Bank Personal Loan Modelling.xlsx", sheet_name = "Data")
df

C:\Users\hp\anaconda3\lib\site-packages\openpyxl\worksheet\_read_only.py:79: UserWarning: Unknown extension is not supported and will be removed
  for idx, row in parser.parse():


ID  Age  Experience  Income  ZIP Code  Family  CCAvg  Education  \
0        1   25           1      49     91107       4    1.6          1   
1        2   45          19      34     90089       3    1.5          1   
2        3   39          15      11     94720       1    1.0          1   
3        4   35           9     100     94112       1    2.7          2   
4        5   35           8      45     91330       4    1.0          2   
...    ...  ...         ...     ...       ...     ...    ...        ...   
4995  4996   29           3      40     92697       1    1.9          3   
4996  4997   30           4      15     92037       4    0.4          1   
4997  4998   63          39      24     93023       2    0.3          3   
4998  4999   65          40      49     90034       3    0.5          2   
4999  5000   28           4      83     92612       3    0.8          1   

      Mortgage  Personal Loan  Securities Account  CD Account  Online  \
0            0              0                   1           0       0   
1            0              0                   1           0       0   
2            0              0                   0           0       0   
3            0              0                   0           0       0   
4            0              0                   0           0       0   
...        ...            ...                 ...         ...     ...   
4995         0              0                   0           0       1   
4996        85              0                   0           0       1   
4997         0              0                   0           0       0   
4998         0              0                   0           0       1   
4999         0              0                   0           0       1   

      CreditCard  
0              0  
1              0  
2              0  
3              0  
4              1  
...          ...  
4995           0  
4996           0  
4997           0  
4998           0  
4999           1  

[5000 rows x 14 columns]

----------

#### Question 8

Give the statistical summary for all the variables in the dataset.

### Answer 8

In [170]:
df2 = df.sort_values(by = ["ID"], ascending = [True])
df3 = df.sort_values(by = ["Age"], ascending = [True])
df4 = df.sort_values(by = ["Experience"], ascending = [True])
df5 = df.sort_values(by = ["Income"], ascending = [True])
df6 = df.sort_values(by = ["ZIP Code"], ascending = [True])
df7 = df.sort_values(by = ["Family"], ascending = [True])
df8 = df.sort_values(by = ["CCAvg"], ascending = [True])
df9 = df.sort_values(by = ["Education"], ascending = [True])
df10 = df.sort_values(by = ["Mortgage"], ascending = [True])

In [171]:
# ID statistical summary
ID_mean = df["ID"].mean()
ID_median = df2["ID"].median()
ID_mode = df["ID"].mode()
ID_kurtosis = df["ID"].kurt()
ID_skew = df["ID"].skew()
ID_max = df2["ID"].max()
ID_min = df2["ID"].min()
ID_range = ID_max - ID_min
ID_coff_range = (ID_max - ID_min)/(ID_max + ID_min)
ID_variance = df["ID"].var()
ID_std =  df["ID"].std()
ID_q1 = df2['ID'].quantile(.25)
ID_q2 = df2['ID'].quantile(.50)
ID_q3 = df2['ID'].quantile(.75)
ID_iqr = ID_q3 - ID_q1

In [172]:
# Age statistical summary
age_mean = df["Age"].mean()
age_median = df3["Age"].median()
age_mode = df["Age"].mode()
age_kurtosis = df["Age"].kurt()
age_skew = df["Age"].skew()
age_max = df3["Age"].max()
age_min = df3["Age"].min()
age_range = age_max - age_min
age_coff_range = (age_max - age_min)/(age_max + age_min)
age_variance = df["Age"].var()
age_std =  df["Age"].std()
age_q1 = df3['Age'].quantile(.25)
age_q2 = df3['Age'].quantile(.50)
age_q3 = df3['Age'].quantile(.75)
age_iqr = age_q3 - age_q1

In [173]:
# Experience statistical summary
experience_mean = df["Experience"].mean()
experience_median = df4["Experience"].median()
experience_mode = df["Experience"].mode()
experience_kurtosis = df["Experience"].kurt()
experience_skew = df["Experience"].skew()
experience_max = df4["Experience"].max()
experience_min = df4["Experience"].min()
experience_range = experience_max - experience_min
experience_coff_range = (experience_max - experience_min)/(experience_max + experience_min)
experience_variance = df["Experience"].var()
experience_std =  df["Experience"].std()
experience_q1 = df4['Experience'].quantile(.25)
experience_q2 = df4['Experience'].quantile(.50)
experience_q3 = df4['Experience'].quantile(.75)
experience_iqr = experience_q3 - experience_q1

In [174]:
# Income statistical summary
income_mean = df["Income"].mean()
income_median = df5["Income"].median()
income_mode = df["Income"].mode()
income_kurtosis = df["Income"].kurt()
income_skew = df["Income"].skew()
income_max = df5["Income"].max()
income_min = df5["Income"].min()
income_range = income_max - income_min
income_coff_range = (income_max - income_min)/(income_max + income_min)
income_variance = df["Income"].var()
income_std =  df["Income"].std()
income_q1 = df5['Income'].quantile(.25)
income_q2 = df5['Income'].quantile(.50)
income_q3 = df5['Income'].quantile(.75)
income_iqr = income_q3 - income_q1

In [175]:
# Zip_Code statistical summary
zip_Code_mean = df["ZIP Code"].mean()
zip_Code_median = df6["ZIP Code"].median()
zip_Code_mode = df["ZIP Code"].mode()
zip_Code_kurtosis = df["ZIP Code"].kurt()
zip_Code_skew = df["ZIP Code"].skew()
zip_Code_max = df6["ZIP Code"].max()
zip_Code_min = df6["ZIP Code"].min()
zip_Code_range = zip_Code_max - zip_Code_min
zip_Code_coff_range = (zip_Code_max - zip_Code_min)/(zip_Code_max + zip_Code_min)
zip_Code_variance = df["ZIP Code"].var()
zip_Code_std =  df["ZIP Code"].std()
zip_Code_q1 = df6['ZIP Code'].quantile(.25)
zip_Code_q2 = df6['ZIP Code'].quantile(.50)
zip_Code_q3 = df6['ZIP Code'].quantile(.75)
zip_Code_iqr = zip_Code_q3 - zip_Code_q1

In [176]:
# Family statistical summary
family_mean = df["Family"].mean()
family_median = df7["Family"].median()
family_mode = df["Family"].mode()
family_kurtosis = df["Family"].kurt()
family_skew = df["Family"].skew()
family_max = df7["Family"].max()
family_min = df7["Family"].min()
family_range = family_max - family_min
family_coff_range = (family_max - family_min)/(family_max + family_min)
family_variance = df["Family"].var()
family_std =  df["Family"].std()
family_q1 = df7['Family'].quantile(.25)
family_q2 = df7['Family'].quantile(.50)
family_q3 = df7['Family'].quantile(.75)
family_iqr = family_q3 - family_q1

In [177]:
# CCAvg statistical summary
CCAvg_mean = df["CCAvg"].mean()
CCAvg_median = df8["CCAvg"].median()
CCAvg_mode = df["CCAvg"].mode()
CCAvg_kurtosis = df["CCAvg"].kurt()
CCAvg_skew = df["CCAvg"].skew()
CCAvg_max = df8["CCAvg"].max()
CCAvg_min = df8["CCAvg"].min()
CCAvg_range = CCAvg_max - CCAvg_min
CCAvg_coff_range = (CCAvg_max - CCAvg_min)/(CCAvg_max + CCAvg_min)
CCAvg_variance = df["CCAvg"].var()
CCAvg_std =  df["CCAvg"].std()
CCAvg_q1 = df8['CCAvg'].quantile(.25)
CCAvg_q2 = df8['CCAvg'].quantile(.50)
CCAvg_q3 = df8['CCAvg'].quantile(.75)
CCAvg_iqr = CCAvg_q3 - CCAvg_q1

In [178]:
# Education statistical summary
Education_mean = df["Education"].mean()
Education_median = df9["Education"].median()
Education_mode = df["Education"].mode()
Education_kurtosis = df["Education"].kurt()
Education_skew = df["Education"].skew()
Education_max = df9["Education"].max()
Education_min = df9["Education"].min()
Education_range = Education_max - Education_min
Education_coff_range = (Education_max - Education_min)/(Education_max + Education_min)
Education_variance = df["Education"].var()
Education_std =  df["Education"].std()
Education_q1 = df9['Education'].quantile(.25)
Education_q2 = df9['Education'].quantile(.50)
Education_q3 = df9['Education'].quantile(.75)
Education_iqr = Education_q3 - Education_q1

In [179]:
# Mortgage statistical summary
Mortgage_mean = df["Mortgage"].mean()
Mortgage_median = df10["Mortgage"].median()
Mortgage_mode = df["Mortgage"].mode()
Mortgage_kurtosis = df["Mortgage"].kurt()
Mortgage_skew = df["Mortgage"].skew()
Mortgage_max = df10["Mortgage"].max()
Mortgage_min = df10["Mortgage"].min()
Mortgage_range = Mortgage_max - Mortgage_min
Mortgage_coff_range = (Mortgage_max - Mortgage_min)/(Mortgage_max + Mortgage_min)
Mortgage_variance = df["Mortgage"].var()
Mortgage_std =  df["Mortgage"].std()
Mortgage_q1 = df10['Mortgage'].quantile(.25)
Mortgage_q2 = df10['Mortgage'].quantile(.50)
Mortgage_q3 = df10['Mortgage'].quantile(.75)
Mortgage_iqr = Mortgage_q3 - Mortgage_q1

In [180]:
print("ID statistical summary-")
print("mean:",ID_mean)
print("median:",ID_median)
print("mode:",ID_mode)
print("kurtosis:",round(ID_kurtosis,2))
print("skew:",round(ID_skew,2))
print("range:",ID_range)
print("variance:",round(ID_variance,2))
print("std:",round(ID_std,2))
print("q1:",ID_q1)
print("q3:",ID_q3)
print("q2:",ID_q2)
print("iqr:",ID_iqr)
print("coff range:",round(ID_coff_range,2))
print("Age statistical summary-")
print("mean:",age_mean)
print("median:",age_median)
print("mode:",age_mode)
print("kurtosis:",round(age_kurtosis,2))
print("skew:",round(age_skew,2))
print("range:",age_range)
print("variance:",round(age_variance,2))
print("std:",round(age_std,2))
print("q1:",age_q1)
print("q3:",age_q3)
print("q2:",age_q2)
print("iqr:",age_iqr)
print("coff range:",round(age_coff_range,2))
print("Experience statistical summary-")
print("mean:",experience_mean)
print("median:",experience_median)
print("mode:",experience_mode)
print("kurtosis:",round(experience_kurtosis,2))
print("skew:",round(experience_skew,2))
print("range:",experience_range)
print("variance:",round(experience_variance,2))
print("std:",round(experience_std,2))
print("q1:",experience_q1)
print("q3:",experience_q3)
print("q2:",experience_q2)
print("iqr:",experience_iqr)
print("coff range:",round(experience_coff_range,2))
print("Income statistical summary-")
print("mean:",income_mean)
print("median:",income_median)
print("mode:",income_mode)
print("kurtosis:",round(income_kurtosis,2))
print("skew:",round(income_skew,2))
print("range:",income_range)
print("variance:",round(income_variance,2))
print("std:",round(income_std,2))
print("q1:",income_q1)
print("q3:",income_q3)
print("q2:",income_q2)
print("iqr:",income_iqr)
print("coff range:",round(income_coff_range,2))
print("Zip_Code statistical summary-")
print("mean:",zip_Code_mean)
print("median:",zip_Code_median)
print("mode:",zip_Code_mode)
print("kurtosis:",round(zip_Code_kurtosis,2))
print("skew:",round(zip_Code_skew,2))
print("range:",zip_Code_range)
print("variance:",round(zip_Code_variance,2))
print("std:",round(zip_Code_std,2))
print("q1:",zip_Code_q1)
print("q3:",zip_Code_q3)
print("q2:",zip_Code_q2)
print("iqr:",zip_Code_iqr)
print("coff range:",round(zip_Code_coff_range,2))
print("Family statistical summary-")
print("mean:",family_mean)
print("median:",family_median)
print("mode:",family_mode)
print("kurtosis:",round(family_kurtosis,2))
print("skew:",round(family_skew,2))
print("range:",family_range)
print("variance:",round(family_variance,2))
print("std:",round(family_std,2))
print("q1:",family_q1)
print("q3:",family_q3)
print("q2:",family_q2)
print("iqr:",family_iqr)
print("coff range:",round(family_coff_range,2))
print("CCAvg statistical summary-")
print("mean:",CCAvg_mean)
print("median:",CCAvg_median)
print("mode:",CCAvg_mode)
print("kurtosis:",round(CCAvg_kurtosis,2))
print("skew:",round(CCAvg_skew,2))
print("range:",CCAvg_range)
print("variance:",round(CCAvg_variance,2))
print("std:",round(CCAvg_std,2))
print("q1:",CCAvg_q1)
print("q3:",CCAvg_q3)
print("q2:",CCAvg_q2)
print("iqr:",CCAvg_iqr)
print("coff range:",round(CCAvg_coff_range,2))
print("Education statistical summary-")
print("mean:",Education_mean)
print("median:",Education_median)
print("mode:",Education_mode)
print("kurtosis:",round(Education_kurtosis,2))
print("skew:",round(Education_skew,2))
print("range:",Education_range)
print("variance:",round(Education_variance,2))
print("std:",round(Education_std,2))
print("q1:",Education_q1)
print("q3:",Education_q3)
print("q2:",Education_q2)
print("iqr:",Education_iqr)
print("coff range:",round(Education_coff_range,2))
print("Mortgage statistical summary-")
print("mean:",Mortgage_mean)
print("median:",Mortgage_median)
print("mode:",Mortgage_mode)
print("kurtosis:",round(Mortgage_kurtosis,2))
print("skew:",round(Mortgage_skew,2))
print("range:",Mortgage_range)
print("variance:",round(Mortgage_variance,2))
print("std:",round(Mortgage_std,2))
print("q1:",Mortgage_q1)
print("q3:",Mortgage_q3)
print("q2:",Mortgage_q2)
print("iqr:",Mortgage_iqr)
print("coff range:",round(Mortgage_coff_range,2))
print("End")

ID statistical summary-
mean: 2500.5
median: 2500.5
mode: 0          1
1          2
2          3
3          4
4          5
        ... 
4995    4996
4996    4997
4997    4998
4998    4999
4999    5000
Name: ID, Length: 5000, dtype: int64
kurtosis: -1.2
skew: 0.0
range: 4999
variance: 2083750.0
std: 1443.52
q1: 1250.75
q3: 3750.25
q2: 2500.5
iqr: 2499.5
coff range: 1.0
Age statistical summary-
mean: 45.3384
median: 45.0
mode: 0    35
Name: Age, dtype: int64
kurtosis: -1.15
skew: -0.03
range: 44
variance: 131.4
std: 11.46
q1: 35.0
q3: 55.0
q2: 45.0
iqr: 20.0
coff range: 0.49
Experience statistical summary-
mean: 20.1046
median: 20.0
mode: 0    32
Name: Experience, dtype: int64
kurtosis: -1.12
skew: -0.03
range: 46
variance: 131.51
std: 11.47
q1: 10.0
q3: 30.0
q2: 20.0
iqr: 20.0
coff range: 1.15
Income statistical summary-
mean: 73.7742
median: 64.0
mode: 0    44
Name: Income, dtype: int64
kurtosis: -0.04
skew: 0.84
range: 216
variance: 2119.1
std: 46.03
q1: 39.0
q3: 98.0
q2: 64.0
iqr: 59

or we can use for statistical summary describe() function

In [224]:
statistical_summary = df[["ID", "Age", "Experience", "Income", "Family","CCAvg", "Education", "Mortgage"]].describe()
statistical_summary

ID          Age   Experience       Income       Family  \
count  5000.000000  5000.000000  5000.000000  5000.000000  5000.000000   
mean   2500.500000    45.338400    20.104600    73.774200     2.396400   
std    1443.520003    11.463166    11.467954    46.033729     1.147663   
min       1.000000    23.000000    -3.000000     8.000000     1.000000   
25%    1250.750000    35.000000    10.000000    39.000000     1.000000   
50%    2500.500000    45.000000    20.000000    64.000000     2.000000   
75%    3750.250000    55.000000    30.000000    98.000000     3.000000   
max    5000.000000    67.000000    43.000000   224.000000     4.000000   

             CCAvg    Education     Mortgage  
count  5000.000000  5000.000000  5000.000000  
mean      1.937913     1.881000    56.498800  
std       1.747666     0.839869   101.713802  
min       0.000000     1.000000     0.000000  
25%       0.700000     1.000000     0.000000  
50%       1.500000     2.000000     0.000000  
75%       2.500000     3.000000   101.000000  
max      10.000000     3.000000   635.000000

-----------

#### Question 9

Give all the measures of central tendency for all the quantitative variables which are continuous and discrete.

### Answer 9

In [181]:
# ID 's measure of central tendency
ID_mean = df["ID"].mean()
ID_median = df["ID"].median()
ID_mode = df["ID"].mode()
# Age 's measure of central tendency
age_mean = df["Age"].mean()
age_median = df["Age"].median()
age_mode = df["Age"].mode()
# Experience 's measure of central tendency
experience_mean = df["Experience"].mean()
experience_median = df["Experience"].median()
experience_mode = df["Experience"].mode()
# Income 's measure of central tendency
income_mean = df["Income"].mean()
income_median = df["Income"].median()
income_mode = df["Income"].mode()
# Zip_Code 's measure of central tendency
zip_Code_mean = df["ZIP Code"].mean()
zip_Code_median = df["ZIP Code"].median()
zip_Code_mode = df["ZIP Code"].mode()
# Family 's measure of central tendency
family_mean = df["Family"].mean()
family_median = df["Family"].median()
family_mode = df["Family"].mode()
# CCAvg 's measure of central tendency
CCAvg_mean = df["CCAvg"].mean()
CCAvg_median = df["CCAvg"].median()
CCAvg_mode = df["CCAvg"].mode()
# Education 's measure of central tendency
Education_mean = df["Education"].mean()
Education_median = df["Education"].median()
Education_mode = df["Education"].mode()
# Mortgage 's measure of central tendency
Mortgage_mean = df["Mortgage"].mean()
Mortgage_median = df["Mortgage"].median()
Mortgage_mode = df["Mortgage"].mode()

In [182]:
print("ID Central Tendency-")
print("mean:",ID_mean)
print("median:",ID_median)
print("mode:",ID_mode)
print("Age Central Tendency-")
print("mean:",age_mean)
print("median:",age_median)
print("mode:",age_mode)
print("Experience Central Tendency-")
print("mean:",experience_mean)
print("median:",experience_median)
print("mode:",experience_mode)
print("Income Central Tendency-")
print("mean:",income_mean)
print("median:",income_median)
print("mode:",income_mode)
print("Zip_Code Central Tendency-")
print("mean:",zip_Code_mean)
print("median:",zip_Code_median)
print("mode:",zip_Code_mode)
print("Family Central Tendency-")
print("mean:",family_mean)
print("median:",family_median)
print("mode:",family_mode)
print("CCAvg Central Tendency-")
print("mean:",CCAvg_mean)
print("median:",CCAvg_median)
print("mode:",CCAvg_mode)
print("Education Central Tendency-")
print("mean:",Education_mean)
print("median:",Education_median)
print("mode:",Education_mode)
print("Mortgage Central Tendency-")
print("mean:",Mortgage_mean)
print("median:",Mortgage_median)
print("mode:",Mortgage_mode)
print("End")

ID Central Tendency-
mean: 2500.5
median: 2500.5
mode: 0          1
1          2
2          3
3          4
4          5
        ... 
4995    4996
4996    4997
4997    4998
4998    4999
4999    5000
Name: ID, Length: 5000, dtype: int64
Age Central Tendency-
mean: 45.3384
median: 45.0
mode: 0    35
Name: Age, dtype: int64
Experience Central Tendency-
mean: 20.1046
median: 20.0
mode: 0    32
Name: Experience, dtype: int64
Income Central Tendency-
mean: 73.7742
median: 64.0
mode: 0    44
Name: Income, dtype: int64
Zip_Code Central Tendency-
mean: 93152.503
median: 93437.0
mode: 0    94720
Name: ZIP Code, dtype: int64
Family Central Tendency-
mean: 2.3964
median: 2.0
mode: 0    1
Name: Family, dtype: int64
CCAvg Central Tendency-
mean: 1.9379133333333332
median: 1.5
mode: 0    0.3
Name: CCAvg, dtype: float64
Education Central Tendency-
mean: 1.881
median: 2.0
mode: 0    1
Name: Education, dtype: int64
Mortgage Central Tendency-
mean: 56.4988
median: 0.0
mode: 0    0
Name: Mortgage, dtype: i

---------

#### Question 10

Can you apply any statistical method to observe any similarity in distribution between age and experience variables?

### Answer 10

In [60]:
# to find the similarity between age and experience we will used the correlation 
correlation_age_experience = df['Age'].corr(df['Experience'])
print("{:.2}".format(correlation_age_experience))

0.99


----------

#### Question 11

Who are the most frequent family members present in this dataset?

### Answer 11

In [62]:
# For finding the most frequent family members in this dataset i will use the mode of central tendency because it is giving the
# number of occurence of a value in a dataset.
family_member = df["Family"].mode()
print("most frequent family members:",family_member)

most frequent family members: 0    1
Name: Family, dtype: int64


-------------

#### Question 12

What is the coefficient of variation you can observe in the ‘Income’ variable?

### Answer 12

In [99]:
# calculate the standard deviation of income
income_std = df["Income"].std()
# calculate mean of income
income_mean = df["Income"].mean()
# calculate the coffecient of variation of income
income_Coff_variance = (income_std/income_mean)*100
print(round(income_Coff_variance,3))

62.398


-----------

#### Question 13

The ‘Mortgage’ variable is having a lot of zeroes. Impute with some business logical value that you feel fit for the data.

### Answer 13

In [137]:
# I got only 6.4% of unique values in mortgage column.
# And 69.24%  are zeros.
percentage_unique_value = df["Mortgage"].nunique()*100/(len(df))
count_zeroes = (df.loc[df["Mortgage"]==0,["Mortgage"]].count())*100/len(df)
print("unique value :", percentage_unique_value,"%")
print("Zero :", count_zeroes)

unique value : 6.94 %
Zero : Mortgage    69.24
dtype: float64


when we find the unique value then i got 6.4% and if i find out the number of zeroes then i got 69.24% it means there is no 
need to replace the number of zeroes with another value but it depends on the business problem if the business think that it 
is beneficial in statistical perspective then i will use mean to replace the zeroes.

Median is best for replacing the zeroes but in this data set median is zero. so, I will use mean. 




In [138]:
mean_mortgage = round(df["Mortgage"].mean(),1)
print(mean_mortgage)
df["Mortgage"] = df["Mortgage"].replace([0], mean_mortgage)
df

56.5


ID  Age  Experience  Income  ZIP Code  Family  CCAvg  Education  \
0        1   25           1      49     91107       4    1.6          1   
1        2   45          19      34     90089       3    1.5          1   
2        3   39          15      11     94720       1    1.0          1   
3        4   35           9     100     94112       1    2.7          2   
4        5   35           8      45     91330       4    1.0          2   
...    ...  ...         ...     ...       ...     ...    ...        ...   
4995  4996   29           3      40     92697       1    1.9          3   
4996  4997   30           4      15     92037       4    0.4          1   
4997  4998   63          39      24     93023       2    0.3          3   
4998  4999   65          40      49     90034       3    0.5          2   
4999  5000   28           4      83     92612       3    0.8          1   

      Mortgage  Personal Loan  Securities Account  CD Account  Online  \
0         56.5              0                   1           0       0   
1         56.5              0                   1           0       0   
2         56.5              0                   0           0       0   
3         56.5              0                   0           0       0   
4         56.5              0                   0           0       0   
...        ...            ...                 ...         ...     ...   
4995      56.5              0                   0           0       1   
4996      85.0              0                   0           0       1   
4997      56.5              0                   0           0       0   
4998      56.5              0                   0           0       1   
4999      56.5              0                   0           0       1   

      CreditCard  
0              0  
1              0  
2              0  
3              0  
4              1  
...          ...  
4995           0  
4996           0  
4997           0  
4998           0  
4999           1  

[5000 rows x 14 columns]

------

#### Question 14

Do customers with credit cards have any impact on their experience and income?

### Answer 14

For analysing the impact of creditcard on their experience and income we divide the data into two categories i.e
1. who are using the creditcard (Y/1)
2. who are not using the creditcards (N/0)

then, i will find the mean and standard deviation and compare both.

In [188]:
creditcard_0 = df.loc[df["CreditCard"]==0,["Income","Experience"]]
creditcard_0

Income  Experience
0         49           1
1         34          19
2         11          15
3        100           9
5         29          13
...      ...         ...
4994      75          40
4995      40           3
4996      15           4
4997      24          39
4998      49          40

[3530 rows x 2 columns]

In [189]:
creditcard_1 = df.loc[df["CreditCard"]==1,["Income","Experience"]]
creditcard_1

Income  Experience
4         45           8
7         22          24
15        22          30
19        21          28
24       152          11
...      ...         ...
4984      98           1
4988      85           8
4990      58          25
4991      92          25
4999      83           4

[1470 rows x 2 columns]

In [200]:
correlation_Y = creditcard_1["Income"].corr(creditcard_1["Experience"])
correlation_Y

-0.033291897228663096

In [202]:
correlation_N = creditcard_0["Experience"].corr(creditcard_0["Income"])
correlation_N

-0.05212627419612275

In both categories Y and N, income and experience have the negative relationship

In [194]:
# Mean
mean_income_0 = creditcard_0["Income"].mean()
mean_income_1 = creditcard_1["Income"].mean()
mean_experience_0 = creditcard_0["Experience"].mean()
mean_experience_1 = creditcard_1["Experience"].mean()
# standard Deviation
std_income_0 = creditcard_0["Income"].std()
std_income_1 = creditcard_1["Income"].std()
std_experience_0 = creditcard_0["Experience"].std()
std_experience_1 = creditcard_1["Experience"].std()
print("Result")
print("mean_income_0 :", round(mean_0,2))
print("mean_income_1 :", round(mean_1,2))
print("mean_experience_0 :", round(mean_1,2))
print("mean_experience_1 :", round(mean_1,2))
print("std__income_0 :", round(std_0,2))
print("std_income_1 :", round(std_0,2))
print("std_experience_0 :", round(std_0,2))
print("std_experience_1 :", round(std_1,2))

Result
mean_income_0 : 73.85
mean_income_1 : 73.6
mean_experience_0 : 73.6
mean_experience_1 : 73.6
std__income_0 : 11.45
std_income_1 : 11.45
std_experience_0 : 11.45
std_experience_1 : 11.52


1. As we can see that the mean of income's N is small high as compared to the income'Y. and standard deviation is same in both cases. this means that the income of N and Y is closer to the mean.

2. And the mean(Experience) of both N and Y is same but standard deviation of N is less than as compare to Y. it means the the N is closer to the mean of experience.

3. After checking the correlation, mean, standard deviation, we can say that the income of Y(who are using the credit cards) impacted by the creditcards. this means Y's income is reduced because of the creditcards.

--------------

#### Question 15

Do you see any outliers in the dataset? If yes what plot is suitable to showcase to the
stakeholders?

### Answer 15

In [64]:
di = df.sort_values(by=["Income"],ascending=[True])
di

ID  Age  Experience  Income  ZIP Code  Family     CCAvg  Education  \
1054  1055   63          38       8     95136       4  0.600000          2   
1197  1198   48          23       8     92866       1  0.400000          3   
939    940   56          32       8     93943       2  0.300000          1   
2591  2592   31           7       8     95131       4  0.700000          2   
2905  2906   64          40       8     94110       2  0.300000          3   
...    ...  ...         ...     ...       ...     ...       ...        ...   
677    678   46          21     204     92780       2  2.800000          1   
526    527   26           2     205     93106       1  6.333333          1   
2988  2989   46          21     205     95762       2  8.800000          1   
4993  4994   45          21     218     91801       2  6.666667          1   
3896  3897   48          24     224     93940       2  6.666667          1   

      Mortgage  Personal Loan  Securities Account  CD Account  Online  \
1054         0              0                   0           0       1   
1197         0              0                   0           0       0   
939          0              0                   0           0       1   
2591         0              0                   0           0       0   
2905         0              0                   0           0       1   
...        ...            ...                 ...         ...     ...   
677          0              0                   0           0       1   
526        271              0                   0           0       0   
2988       181              0                   1           0       1   
4993         0              0                   0           0       1   
3896         0              0                   0           1       1   

      CreditCard  
1054           0  
1197           1  
939            1  
2591           0  
2905           0  
...          ...  
677            0  
526            1  
2988           0  
4993           0  
3896           1  

[5000 rows x 14 columns]

In [116]:
inc = (di.loc[di["Income"]==8].count())/len(di)
inc

ID                    0.0046
Age                   0.0046
Experience            0.0046
Income                0.0046
ZIP Code              0.0046
Family                0.0046
CCAvg                 0.0046
Education             0.0046
Mortgage              0.0046
Personal Loan         0.0046
Securities Account    0.0046
CD Account            0.0046
Online                0.0046
CreditCard            0.0046
dtype: float64

In [65]:
income_q1 = di['Income'].quantile(.25)
income_q3 = di['Income'].quantile(.75)
income_iqr = income_q3 - income_q1
lower_bound = income_q1 - (1.5*income_iqr)
higher_bound = income_q3 + (1.5*income_iqr)
income_outlier = di.loc[(di["Income"]>higher_bound) | (di["Income"]<lower_bound)]
income_outlier

ID  Age  Experience  Income  ZIP Code  Family     CCAvg  Education  \
2087  2088   51          27     188     94305       2  6.900000          2   
4655  4656   33           7     188     95054       2  7.000000          2   
1719  1720   36          12     188     91304       2  6.500000          1   
4309  4310   34           8     188     94025       1  2.900000          3   
1097  1098   50          24     188     92007       3  1.300000          1   
...    ...  ...         ...     ...       ...     ...       ...        ...   
677    678   46          21     204     92780       2  2.800000          1   
526    527   26           2     205     93106       1  6.333333          1   
2988  2989   46          21     205     95762       2  8.800000          1   
4993  4994   45          21     218     91801       2  6.666667          1   
3896  3897   48          24     224     93940       2  6.666667          1   

      Mortgage  Personal Loan  Securities Account  CD Account  Online  \
2087       343              1                   0           0       1   
4655       581              1                   0           0       0   
1719         0              0                   0           0       1   
4309         0              1                   0           0       1   
1097       184              1                   0           0       0   
...        ...            ...                 ...         ...     ...   
677          0              0                   0           0       1   
526        271              0                   0           0       0   
2988       181              0                   1           0       1   
4993         0              0                   0           0       1   
3896         0              0                   0           1       1   

      CreditCard  
2087           0  
4655           0  
1719           1  
4309           0  
1097           1  
...          ...  
677            0  
526            1  
2988           0  
4993           0  
3896           1  

[96 rows x 14 columns]

In [73]:
df1 = df.sort_values(by = ["CCAvg"], ascending = [True])

In [74]:
# outlier of CCAvg
CCAvg_q1 = df1['CCAvg'].quantile(.25)
CCAvg_q3 = df1['CCAvg'].quantile(.75)
CCAvg_iqr = CCAvg_q3 - CCAvg_q1
lower_bound = CCAvg_q1 - (1.5*CCAvg_iqr)
higher_bound = CCAvg_q3 + (1.5*CCAvg_iqr)
CCAvg_outlier = df1.loc[(df1["CCAvg"]>higher_bound) | (df1["CCAvg"]<lower_bound)]
CCAvg_outlier

ID  Age  Experience  Income  ZIP Code  Family      CCAvg  Education  \
4980  4981   29           5     135     95762       3   5.300000          1   
351    352   28           4     155     92182       4   5.300000          2   
1768  1769   43          18     128     92093       4   5.300000          1   
4470  4471   44          20     111     91911       2   5.300000          2   
3368  3369   45          18     163     94720       3   5.333333          2   
...    ...  ...         ...     ...       ...     ...        ...        ...   
3822  3823   63          33     178     91768       4   9.000000          3   
3943  3944   61          36     188     91360       1   9.300000          2   
2101  2102   35           5     203     95032       1  10.000000          3   
787    788   45          15     202     91380       3  10.000000          3   
2337  2338   43          16     201     95054       1  10.000000          2   

      Mortgage  Personal Loan  Securities Account  CD Account  Online  \
4980         0              1                   0           1       1   
351          0              1                   0           0       1   
1768        84              1                   0           0       0   
4470         0              1                   0           0       1   
3368         0              1                   0           1       0   
...        ...            ...                 ...         ...     ...   
3822         0              1                   0           0       0   
3943         0              1                   0           0       0   
2101         0              1                   0           0       0   
787          0              1                   0           0       0   
2337         0              1                   0           0       0   

      CreditCard  
4980           1  
351            0  
1768           0  
4470           0  
3368           1  
...          ...  
3822           0  
3943           0  
2101           0  
787            0  
2337           1  

[324 rows x 14 columns]

In [100]:
df1 = df.sort_values(by = ["Mortgage"], ascending = [True])
df1

ID  Age  Experience  Income  ZIP Code  Family  CCAvg  Education  \
0        1   25           1      49     91107       4    1.6          1   
3093  3094   29           5      34     90717       4    0.4          2   
3095  3096   49          25      43     94109       1    1.6          2   
3096  3097   43          18     179     94108       3    1.2          1   
3097  3098   58          32      44     92182       3    2.2          3   
...    ...  ...         ...     ...       ...     ...    ...        ...   
4842  4843   49          23     174     95449       3    4.6          2   
1783  1784   53          27     192     94720       1    1.7          1   
4812  4813   29           4     184     92126       4    2.2          3   
303    304   49          25     195     95605       4    3.0          1   
2934  2935   37          13     195     91763       2    6.5          1   

      Mortgage  Personal Loan  Securities Account  CD Account  Online  \
0            0              0                   1           0       0   
3093         0              0                   0           0       0   
3095         0              0                   0           0       1   
3096         0              1                   1           1       1   
3097         0              0                   0           0       0   
...        ...            ...                 ...         ...     ...   
4842       590              1                   0           0       0   
1783       601              0                   0           0       1   
4812       612              1                   0           0       1   
303        617              1                   0           0       0   
2934       635              0                   0           0       1   

      CreditCard  
0              0  
3093           1  
3095           1  
3096           0  
3097           0  
...          ...  
4842           0  
1783           0  
4812           0  
303            0  
2934           0  

[5000 rows x 14 columns]

In [104]:
# outlier of Mortgage
Mortgage_q1 = df1['Mortgage'].quantile(.25)
Mortgage_q3 = df1['Mortgage'].quantile(.75)
Mortgage_iqr = Mortgage_q3 - Mortgage_q1
lower_bound = Mortgage_q1 - (1.5*Mortgage_iqr)
higher_bound = Mortgage_q3 + (1.5*Mortgage_iqr)
Mortgage_outlier = df1.loc[(df1["Mortgage"]>higher_bound) | (df1["Mortgage"]<lower_bound)]
Mortgage_outlier

ID  Age  Experience  Income  ZIP Code  Family  CCAvg  Education  \
3948  3949   37          12     123     94304       4    3.1          2   
3138  3139   36          11     103     93555       1    4.6          1   
3950  3951   38          14      62     94143       1    1.5          3   
2159  2160   61          35      99     94085       1    4.8          3   
4483  4484   54          28     155     90095       1    1.0          1   
...    ...  ...         ...     ...       ...     ...    ...        ...   
4842  4843   49          23     174     95449       3    4.6          2   
1783  1784   53          27     192     94720       1    1.7          1   
4812  4813   29           4     184     92126       4    2.2          3   
303    304   49          25     195     95605       4    3.0          1   
2934  2935   37          13     195     91763       2    6.5          1   

      Mortgage  Personal Loan  Securities Account  CD Account  Online  \
3948       253              1                   0           1       1   
3138       255              0                   0           0       1   
3950       255              0                   0           0       1   
2159       255              1                   0           0       0   
4483       256              0                   0           0       1   
...        ...            ...                 ...         ...     ...   
4842       590              1                   0           0       0   
1783       601              0                   0           0       1   
4812       612              1                   0           0       1   
303        617              1                   0           0       0   
2934       635              0                   0           0       1   

      CreditCard  
3948           1  
3138           0  
3950           0  
2159           1  
4483           1  
...          ...  
4842           0  
1783           0  
4812           0  
303            0  
2934           0  

[291 rows x 14 columns]

----------

#### Question 16

Give us the deciles values of the variable ‘Income’ in the dataset.

### Answer 16

for calculating deciles values we need to sort the income data

In [226]:
df1 = df.sort_values(by = ["Income"], ascending = [True])
df1

ID  Age  Experience  Income  ZIP Code  Family     CCAvg  Education  \
1054  1055   63          38       8     95136       4  0.600000          2   
1197  1198   48          23       8     92866       1  0.400000          3   
939    940   56          32       8     93943       2  0.300000          1   
2591  2592   31           7       8     95131       4  0.700000          2   
2905  2906   64          40       8     94110       2  0.300000          3   
...    ...  ...         ...     ...       ...     ...       ...        ...   
677    678   46          21     204     92780       2  2.800000          1   
526    527   26           2     205     93106       1  6.333333          1   
2988  2989   46          21     205     95762       2  8.800000          1   
4993  4994   45          21     218     91801       2  6.666667          1   
3896  3897   48          24     224     93940       2  6.666667          1   

      Mortgage  Personal Loan  Securities Account  CD Account  Online  \
1054         0              0                   0           0       1   
1197         0              0                   0           0       0   
939          0              0                   0           0       1   
2591         0              0                   0           0       0   
2905         0              0                   0           0       1   
...        ...            ...                 ...         ...     ...   
677          0              0                   0           0       1   
526        271              0                   0           0       0   
2988       181              0                   1           0       1   
4993         0              0                   0           0       1   
3896         0              0                   0           1       1   

      CreditCard  
1054           0  
1197           1  
939            1  
2591           0  
2905           0  
...          ...  
677            0  
526            1  
2988           0  
4993           0  
3896           1  

[5000 rows x 14 columns]

In [227]:
df1["Income"].quantile([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9])

0.1     22.0
0.2     33.0
0.3     42.0
0.4     52.0
0.5     64.0
0.6     78.0
0.7     88.3
0.8    113.0
0.9    145.0
Name: Income, dtype: float64

-----------

In [186]:
df2 = df.sort_values(by = ["ID"], ascending = [True])
df3 = df.sort_values(by = ["Age"], ascending = [True])
df4 = df.sort_values(by = ["Experience"], ascending = [True])
df5 = df.sort_values(by = ["Income"], ascending = [True])
df6 = df.sort_values(by = ["ZIP Code"], ascending = [True])
df7 = df.sort_values(by = ["Family"], ascending = [True])
df8 = df.sort_values(by = ["CCAvg"], ascending = [True])
df9 = df.sort_values(by = ["Education"], ascending = [True])
df10 = df.sort_values(by = ["Mortgage"], ascending = [True])

#### Question 17

Give the IQR of all the variables which are quantitative and continuous.

### Answer 17

In [185]:
# ID's IQR
ID_q1 = df2['ID'].quantile(.25)
ID_q3 = df2['ID'].quantile(.75)
ID_iqr = ID_q3 - ID_q1

# Age's IQR
age_q1 = df3['Age'].quantile(.25)
age_q3 = df3['Age'].quantile(.75)
age_iqr = age_q3 - age_q1

# Experience's IQR
experience_q1 = df4['Experience'].quantile(.25)
experience_q3 = df4['Experience'].quantile(.75)
experience_iqr = experience_q3 - experience_q1

# Income's IQR
income_q1 = df5['Income'].quantile(.25)
income_q3 = df5['Income'].quantile(.75)
income_iqr = income_q3 - income_q1

# Zip Code's IQR
zip_Code_q1 = df6['ZIP Code'].quantile(.25)
zip_Code_q3 = df6['ZIP Code'].quantile(.75)
zip_Code_iqr = zip_Code_q3 - zip_Code_q1

# Family's IQR
family_q1 = df7['Family'].quantile(.25)
family_q3 = df7['Family'].quantile(.75)
family_iqr = family_q3 - family_q1

# CCAvg's IQR
CCAvg_q1 = df8['CCAvg'].quantile(.25)
CCAvg_q3 = df8['CCAvg'].quantile(.75)
CCAvg_iqr = CCAvg_q3 - CCAvg_q1

# Education's IQR
Education_q1 = df9['Education'].quantile(.25)
Education_q3 = df9['Education'].quantile(.75)
Education_iqr = Education_q3 - Education_q1

# Mortgage's IQR
Mortgage_q1 = df10['Mortgage'].quantile(.25)
Mortgage_q3 = df10['Mortgage'].quantile(.75)
Mortgage_iqr = Mortgage_q3 - Mortgage_q1

In [184]:
print("Age IQR:",age_iqr)
print("Experience IQR:",experience_iqr)
print("Income IQR:",income_iqr)
print("Zip code IQR:",zip_Code_iqr)
print("Family IQR:",family_iqr)
print("CCAvg IQR:",CCAvg_iqr)
print("Education IQR:",Education_iqr)
print("Mortgage IQR:",Mortgage_iqr)

Age IQR: 20.0
Experience IQR: 20.0
Income IQR: 59.0
Zip code IQR: 2697.0
Family IQR: 2.0
CCAvg IQR: 1.8
Education IQR: 2.0
Mortgage IQR: 101.0


-----------

#### Question 18

Do the higher-income holders spend more on credit cards?

### Answer 18

In [225]:
income_grouped = df.groupby("Income")["CCAvg"].mean()
income_grouped

Income
8      0.428986
9      0.384615
10     0.481159
11     0.402469
12     0.522222
         ...   
203    9.400000
204    5.366667
205    7.566667
218    6.666667
224    6.666667
Name: CCAvg, Length: 162, dtype: float64

1. Now, we can see that the income group of 203 is expanding on an average 9.4 on creditcards 
2. And the income group of 224 is expanding on an average 6.6 on creditcards.
3. so, we can not say that if the higher income holders spend more on credits.

--------

#### Question 19

Do customers using bank internet facilities paid higher?

### Answer 19

In [10]:
# I am considering online and CCAvg
# who are using the bank facilities
bank_int_fac_y = df.loc[df["Online"]==1,["CCAvg"]]
bank_int_fac_y

CCAvg
5       0.4
6       1.5
8       0.6
11      0.1
13      2.5
...     ...
4994    2.0
4995    1.9
4996    0.4
4998    0.5
4999    0.8

[2984 rows x 1 columns]

1. y = who are using bank internet facility)
2. n = who are not using bank internet facility)

In [8]:
# who are not using bank facilities
bank_int_fac_n = df.loc[df["Online"]==0, ["CCAvg"]]
bank_int_fac_n

CCAvg
0       1.6
1       1.5
2       1.0
3       2.7
4       1.0
...     ...
4988    2.5
4990    2.0
4991    1.9
4992    0.5
4997    0.3

[2016 rows x 1 columns]

In [12]:
mean_y = bank_int_fac_y["CCAvg"].mean()
print(round(mean_y,2))
mean_n = bank_int_fac_n["CCAvg"].mean()
print(round(mean_n,2))

1.93
1.95


In [13]:
std_y = bank_int_fac_y["CCAvg"].std()
print(round(std_y,2))
std_n = bank_int_fac_n["CCAvg"].std()
print(round(std_n,2))

1.75
1.74


As we can see that y mean is less than n mean which means on average n is paid higher amount as compared to y but the std of y is more than std of n which means the paid amont is widely spread from the mean as compared to n obviously there is a little bit difference between mean and standard deviation of both y and n. than we can say that y(who are using bank internet facility) is paying higher

---------------

#### Question 20

Calculate the Z score of the ‘income’ variable.

### Answer 20

In [140]:
# calculate mean of income
income_mean = df["Income"].mean()
# calculate standard deviation of income
income_std = df["Income"].std()
# calculate Z score of income
z_score_income = ((df["Income"]-income_mean)/income_std)
print(round(z_score_income,2))

0      -0.54
1      -0.86
2      -1.36
3       0.57
4      -0.63
        ... 
4995   -0.73
4996   -1.28
4997   -1.08
4998   -0.54
4999    0.20
Name: Income, Length: 5000, dtype: float64


---------------------------------
---------------------------
---------------------------------